src/main/kotlin/racing/service/RacingService.kt

```diff
@@ -13,28 +13,21 @@ class RacingService {
     fun racing(participates: ArrayDeque<String>, roundCount: Int): RacingHistory {
         val racingHistory = RacingHistory()
         racingHistory.appendRound(ready(participates))
-        (1..roundCount).forEach { _ ->
+        (1 until roundCount).forEach { _ ->
             racingHistory.appendRound(nextRound(racingHistory.lastRound()))
         }
         return racingHistory
     }
 
     private fun ready(participates: ArrayDeque<String>): RoundHistory {
         return RoundHistory(
-            ArrayDeque(
-                participates.map {
-                    Car(it, 0)
-                }
-            )
+            participates.map { Car(it, 0) }
         )
     }
 
     private fun nextRound(preRoundHistory: RoundHistory): RoundHistory {
         return RoundHistory(
-            ArrayDeque(
-                preRoundHistory.result
-                    .map { it.move(strategy).copy() }
-            )
+            preRoundHistory.results.map { it.move(strategy).copy() }
```

Comment:
result라는 내부 프로퍼티를 꺼내서 직접 연산을 하고 있네요!
```suggestion
            preRoundHistory.map { it.move(strategy).copy() }
```
이렇게 해도 괜찮을 것 같아요!